### ISSUE TITLE
The URL is time-sensitive. I want to request the server to obtain the URL every time I play. Is this possible?

### ISSUE THREAD

**Issue by 2697a** [media-kit/media-kit#715]
The URL is time-sensitive. I want to request the server to obtain the URL every time I play. Is this possible?

**Comment by alexmercerind** [comment]
I'm really unable to understand, but does it really relate to plugin? You should make some own API call, no?

**Comment by zezo357** [comment]
> The URL is time-sensitive. I want to request the server to obtain the URL every time I play. Is this possible?

i agree with alex, i think you should maintain that behaviour in your code, by getting the new url then providing it to the player

**Comment by 2697a** [comment]
I want to use the playlist of the plug-in, because the timeliness of the song address is relatively short, maybe after the first song in the list is played, the URL of the other songs will be invalid, so I want to dynamically set the URL of the song, for example, when the next song is about to play, I can get the URL of the next song to play, and then play, if I play one by one, I need to manage the playlist by myself, and I can't use the playlist in the plug-in

**Comment by zezo357** [comment]
I think you can get the urls before playing each item, like if the track is about to end let's say 10 seconds remaining, then you update the next track url ?

**Comment by alexmercerind** [comment]
@2697a 

I understand your use-case. I don't think this situation can be handled by media_kit solely or any other media playback library. I had a similar problem few days ago & my solution was to run a local HTTP server & redirect to original URL using status code 301/302. This way playlists (or any expiring URL) can also be handled effectively.

<hr>

I have invited you to my personal repository. I think it will serve you as a sample. It is un-related to this project & I cannot provide further details in this specific regard due to lack of time.

Thanks!
